server2/Arcadia.Assistant/Arcadia.Assistant.Sharepoint/SharepointSynchronizator.cs

```diff
@@ -0,0 +1,170 @@
+                        // synchronize workouts for selected department
+                        workoutsSync.SynchronizeItems(calendar, departmentEmployees, employeeWorkoutsValues, storageItems, cancellationToken),
+                        // synchronize vacations for selected department
+                        sickLeavesSync.SynchronizeItems(calendar, departmentEmployees, employeeSickLeavesValues, storageItems, cancellationToken));
+                }
+            }
+        }
+
+        #endregion
+
+        #region internal class
+
+        private class EmployeeVacationsSynchronization : SharepointItemSynchronization<VacationDescription>
+        {
+            public EmployeeVacationsSynchronization(IExternalStorage externalStorage, ILogger? logger = null)
+                : base(externalStorage, logger)
+            {
+            }
+
+            protected override string ItemEventType { get; } = CalendarEventTypes.Vacation;
+
+            protected override DatesPeriod GetItemDatePeriod(VacationDescription item)
+            {
+                return new DatesPeriod(item.StartDate.Date, item.EndDate.Date);
+            }
+
+            protected override EmployeeId GetItemEmployeeId(VacationDescription item)
+            {
+                return item.EmployeeId;
+            }
+        }
+
+        private class EmployeeWorkoutsSynchronization : SharepointItemSynchronization<WorkHoursChange>
+        {
+            public EmployeeWorkoutsSynchronization(IExternalStorage externalStorage, ILogger? logger = null)
+                : base(externalStorage, logger)
+            {
+            }
+
+            protected override string ItemEventType { get; } = CalendarEventTypes.Workout;
+
+            protected override DatesPeriod GetItemDatePeriod(WorkHoursChange item)
+            {
+                return new DatesPeriod(item.Date, item.Date);
+            }
+
+            protected override EmployeeId GetItemEmployeeId(WorkHoursChange item)
+            {
+                return item.EmployeeId;
+            }
+        }
+
+        private class EmployeeSickLeavesSynchronization : SharepointItemSynchronization<SickLeaveDescription>
+        {
+            public EmployeeSickLeavesSynchronization(IExternalStorage externalStorage, ILogger? logger = null)
+                : base(externalStorage, logger)
+            {
+            }
+
+            protected override string ItemEventType { get; } = CalendarEventTypes.Sickleave;
+
+            protected override DatesPeriod GetItemDatePeriod(SickLeaveDescription item)
+            {
+                return new DatesPeriod(item.StartDate.Date, item.EndDate.Date);
+            }
+
+            protected override EmployeeId GetItemEmployeeId(SickLeaveDescription item)
+            {
+                return item.EmployeeId;
+            }
+        }
+
+        #endregion
+
+        #region variables
+
+        private readonly ISickLeaves sickLeaves;
```

Comment:
By convention fields should be placed before all constructors and inner classes should be located in the very end of the class.

Reply:
Moved